src/pybind/cephfs/cephfs.pyx

```diff
@@ -2832,3 +2837,186 @@ cdef class LibCephFS(object):
 
         finally:
            free(buf)
+
+
+    def rmtree(self, trash_path, should_cancel):
```

Comment:
I would suggest two modes for this:

`suppress_errors = False` (default False)

The ceph-mgr will use `suppress_errors = True` to try to get as far as possible. There should be a return value indicating whether any errors were encountered (perhaps, a count instead of a specific error).

If it is `False`, then immediately raise an exception and quit `rmtree`.